aws/tools/cleanup-ssm/src/cleanup.rs

```diff
@@ -0,0 +1,151 @@
+use crate::SsmClient;
+
+pub async fn delete_parameters_in_batches<C: SsmClient>(
+    client: &C,
+    parameters_to_delete: Vec<String>,
+) -> Result<(usize, usize), aws_sdk_ssm::Error> {
+    let mut total_deleted = 0;
+    let mut total_failed = 0;
+
+    for chunk in parameters_to_delete.chunks(10) {
+        let (deleted, failed) = client.delete_parameters(chunk.to_vec()).await?;
+
+        total_deleted += deleted.len();
+        total_failed += failed.len();
+
+        println!("Deleted {} parameters", deleted.len());
+        if !failed.is_empty() {
+            println!("Failed to delete {} parameters", failed.len());
+        }
+    }
+
+    Ok((total_deleted, total_failed))
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Is it a rust thing to put tests in the same file as the code?  This feels really weird

Reply:
Yup! For things like unit tests it's expected, integration tests live in `<ROOT>/tests/`